minimesos/src/main/java/com/containersol/minimesos/docker/DockerContainersUtil.java

```diff
@@ -102,9 +103,24 @@ public void remove() {
      * Removes all containers in the util object
      */
     public DockerContainersUtil kill() {
+        return kill(false);
+    }
+
+    /**
+     * Removes all containers in the util object
+     *
+     * @param ignoreFailure - use <code>true</code> if you expect containers might be stopped by this time
+     */
+    public DockerContainersUtil kill(boolean ignoreFailure) {
         if (containers != null) {
             for (Container container : containers) {
-                dockerClient.killContainerCmd(container.getId()).exec();
+                try {
+                    dockerClient.killContainerCmd(container.getId()).exec();
+                } catch (DockerException failure) {
+                    if(!ignoreFailure) {
```

Comment:
The "not running container" exception is `com.github.dockerjava.api.InternalServerErrorException: Cannot kill container` - there is no special exception for attempts to kill not-running container. My choices were:
* start parsing the error messages;
* or catch more generic exception and assume the container is not running. I guess other errors would be detected during either attempt to lookup containers and the following attempt to remove the container